transport/tchannel/options.go

```diff
@@ -220,3 +221,10 @@ func WithNativeTChannelMethods(nativeMethods NativeTChannelMethods) TransportOpt
 		option.nativeTChannelMethods = nativeMethods
 	}
 }
+
+// DisableRpcServiceHeader disables adding the $rpc$-service header
+func DisableRpcServiceHeader() TransportOption {
+	return func(option *transportOptions) {
+		option.disableRpcServiceHeader = true
+	}
+}
```

Comment:
Using a negative setter for bools is usually a bad pattern.  Can we invert this and document the default value if not leveraged?

This way, `option.disableRpcServiceHeader = false` avoids a double negative.

For example,
```
func AddRpcServiceHeader(val bool) TransportOption
```
sidenote: `Add...`, `Use...` or `Enable...` or any equivalent such naming.

Reply:
The reason I chose this way is so that I can use the zero value to effectively keep the existing functionality easily. But if we prefer I can make the change and carefully ensure we set the value to true where ever it is instantiated.

Reply:
IMO, Having exclude prefix more suitable in this case due to its opt-out nature. We want the services to explicitly pass this exclude option to disable the service header in the response.

Reply:
What @witriew is referring to is the internal data structure. External api will still be exclude. Internally we can still have struct field as `addServiceHeaderInResponse`. We just need to ensure that field is set to true by default where Transport is created.

Reply:
oh Sorry @witriew  is referring to the api. Yea, the api should be Exclude. Internally we can choose the struct to have a field to addServiceHeaderInResponse set to true.

Reply:
I'm referring to both the API and the internal param.

> IMO, Having exclude prefix more suitable in this case due to its opt-out nature. We want the services to explicitly pass this exclude option to disable the service header in the response.

The default internally can be whatever we choose it to be, as long as it's documented what the behavior is.  The API provided doesn't necessarily need to expose what the internal setting was, but rather, its simplicity when it's being used should be considered, as in, setting a `disable` to `false` feels awkward.

Reply:
@witriew  - just to conclude this - can you be explicit in what you are suggesting? API and the internal field please.

Reply:
Btw, please note that we are not setting `option.disableRpcServiceHeader = false` instead we are doing this `option.disableRpcServiceHeader = true`

Reply:
yea we should be not setting the value to `false`. The zero value is already `false` and we should utilize that.

Reply:
> @witriew - just to conclude this - can you be explicit in what you are suggesting? API and the internal field please.

Yah - well, the internal field should match what the API is setting and mentioned briefly [above](https://github.com/yarpc/yarpc-go/pull/2120#discussion_r797207606):
```
func AddRpcServiceHeader(val bool) TransportOption
```
taking in the value to set it to, but defaulting to `true` internally.

But if @AllenLuUber is fine with this, it's his call.

Reply:
yea `DisableRpcServiceHeader ` sounds good enough.